master/internal/stream/trials.go

```diff
@@ -121,17 +129,17 @@ func TrialCollectStartupMsgs(known string, spec TrialSubscriptionSpec, ctx conte
 		}
 	}
 
-	// step 4: emit deletions and udpates to the client
-	out = append(out, newDeletedMsg(TrialsDeleteKey, missing))
+	// step 4: emit deletions and updates to the client
+	out = append(out, stream.NewDeleteMsg(TrialsDeleteKey, missing))
 	for _, msg := range trialMsgs {
-		out = append(out, msg.UpsertMsg())
+		out = append(out, stream.NewUpsertMsg(msg, "trial"))
```

Comment:
there's a typing issue here.

Before, this function emitted `PreparedMessage`s, which always indicated this was the prepared message that's ready to ship.  That was effectively `PrepareFn(stream.UpsertMsg{})`.

Now it's emitting more like the equivalent of a bare `stream.UpsertMsg{}`.  Is this intentional?

On the one hand, if NewUpsertMsg would json-marshal to the right thing, and there's no need for caching in this codepath, it might be ok.  You'd have to catch non-PreparedMessage values that get sent to the websocket.

On the other hand, I think it's a little confusing that the "preparedness level" of these `interface{}` messages differs between the startup codepath and the live-stream codepath.

Reply:
yeah this is something I'm still thinking about. My first shot at this is incoming - I've made a `PreparableMsg` interface that the upsert and delete messages embed, and I feel like that should help clarify things.